modelling/src/main/java/org/axonframework/modelling/saga/SagaLifecycle.java

```diff
@@ -108,6 +119,8 @@ public static void end() {
      */
     protected abstract void doAssociateWith(AssociationValue associationValue);
 
+    protected abstract AssociationValues getAssociationValues();
```

Comment:
Seems like a doable thing to me! Would you mind making an actual implementation of this?